### PR TITLE
infra-deployments: skip CI for stage-only components

### PR DESCRIPTION
**Pull request by Michkov**
Skip components which are not deployed on development/testing clusters.

### CODE REVIEW

**Comment by amfred** [comment]
A little more background if we come back to this later:
https://github.com/redhat-appstudio/infra-deployments/blob/main/argo-cd-apps/overlays/development/delete-applications.yaml 
Preview mode is used by e2e-tests, and that is using the development overlay. Therefore changes to these components are not currently in the CI flow. We could revisit that later but it's the current state of things.

**Review by rhopp — APPROVED** [review]

File: ci-operator/jobs/redhat-appstudio/infra-deployments/redhat-appstudio-infra-deployments-main-presubmits.yaml [hunk 1]
```diff
@@ -16,7 +16,7 @@ presubmits:
     name: pull-ci-redhat-appstudio-infra-deployments-main-appstudio-e2e-tests
     optional: true
     rerun_command: /test appstudio-e2e-tests
-    skip_if_only_changed: ^docs/|^\.github/|^\.tekton/|\.md$|^(?:.*/)?(?:\.gitignore|OWNERS|PROJECT|LICENSE)$
+    skip_if_only_changed: ^docs/|^\.github/|^\.tekton/|^components/(authentication|quality-dashboard|hac-pact-broker|monitoring-workload-in-cluster|tekton-ci)/|\.md$|^(?:.*/)?(?:\.gitignore|OWNERS|PROJECT|LICENSE)$
```
Comment (bamachrn) [inline ci-operator/jobs/redhat-appstudio/infra-deployments/redhat-appstudio-infra-deployments-main-presubmits.yaml:19]: these are folders? or argcd applications?

**Review by bamachrn — CHANGES_REQUESTED** [review]
monitoring directory needs to be changed to `components/monitoring`

**Review by bamachrn — APPROVED** [review]
looks good to me

**Review by amfred — APPROVED** [review]

**Pull request merged**